dj/models/node.py

```diff
@@ -70,6 +70,34 @@ class NodeType(str, enum.Enum):
     DIMENSION = "dimension"
 
 
+class NodeMode(str, enum.Enum):
+    """
+    Node mode.
+
+    A node can be in one of the following modes:
+
+    1. PUBLISHED nodes must be valid and not cause any child nodes to be invalid
+    2. DRAFT nodes can be invalid, have invalid parents, and include dangling references
+    """
+
+    PUBLISHED = "published"
+    DRAFT = "draft"
+
+
+class NodeStatus(str, enum.Enum):
+    """
+    Node status.
+
+    A node can have one of the following statuses:
+
+    1. VALID nodes - All references to other nodes and node columns are valid
```

Comment:
nit: s/VALID nodes/VALID, same for INVALID. It reads better imo.

Reply:
Updated. Agreed it reads much better!